java/deepgram_audio_summary/Index.java

```diff
@@ -0,0 +1,171 @@
+        if(errorResponse!=null){
+            return errorResponse;
+        }
+
+        // Validate the requested payload (URL)
+        String payloadString=req.getPayload().toString();
+        Map<String, Object> payload=gson.fromJson(payloadString,Map.class);
+
+        errorResponse=validatePayload(payload,res);
+        if(errorResponse!=null){
+            return errorResponse;
+        }
+
+        // Get file url from payload
+        String fileurl=payload.get("fileUrl").toString();
+
+        // Name API key is not empty
+        String apiKeyVariable="DEEPGRAM_SECRET_KEY";
+
+        errorResponse=checkEmptyAPIKey(req,res,apiKeyVariable);
+        if(errorResponse!=null){
+            return errorResponse;
+        }
+
+        String apiKey=req.getVariables().get(apiKeyVariable).toString();
+
+        String deepgramData="";
+        Map<String, Object> responseData=new HashMap<>();
+
+        try{
+            deepgramData=generateSummary(fileurl,apiKey);
+        }catch(Exception e){
+            responseData.put("success",false);
+            responseData.put("message","Something went wrong while generating the summary, please check with the developers. Error: "+e.getMessage());//TODO
+        return res.json(responseData);
+        }
+        responseData.put("success",true);
+        responseData.put("deepgramData",deepgramData);
+
+        return res.json(responseData);
+}
+
+
+/**
+ * This method will send a POST request to the specified endpoint and return the Deepgram's response
+ *
+ * @param requestBody is the Request Body for the POST request containing the URL to the wav file to Summarize
+ * @param apiKey is the access token used by Deepgram to summarize
+ * @return the Deepgram's response to the POST request
+ * @throws Exception in case of malformed URL, I/O exception, etc.
+ */
+private String generateSummary(String requestBody,String apiKey)throws Exception{
+        String endpointUrl="https://api.deepgram.com/v1/listen?summarize=true&punctuate=true";
+        URL url=new URL(endpointUrl);
+        HttpURLConnection con=(HttpURLConnection)url.openConnection();
+
+        con.setRequestMethod("POST");
+        con.setRequestProperty("Content-Type","application/json");
+        con.setRequestProperty("Authorization","Token "+apiKey);
+
+        con.setDoOutput(true);
+
+        //prepare request body
+        requestBody="{\"url\":\""+requestBody+"\"}";
+
+        OutputStream os=con.getOutputStream();
+        byte[] input=requestBody.getBytes("utf-8");
+        os.write(input,0,input.length);
+
+        StringBuilder response=new StringBuilder();
+
+        BufferedReader br=new BufferedReader(new InputStreamReader(con.getInputStream(),"utf-8"));
+        String responseLine=null;
+        while((responseLine=br.readLine())!=null){
+            response.append(responseLine.trim());
+        }
+
+        br.close();
+        con.disconnect();
+
+        return response.toString();
```

Comment:
To be consistent with the other implementations of this, would you please return the response as a JSON object (probably a `Map<String, Object>`)?

Reply:
I'm sorry I don't understand this very well. This code that you linked is the return about "generateSummary" method. It return a string and then the string will be put into a json for the function response. So, if I trasform this in a json the function response will be something like> {"success": true, "deepgramData":{"summary":"[REAL_DEEPGRAM_RESPONSE]"}}.